anesthetic/weighted_pandas.py

```diff
@@ -143,6 +154,18 @@ def compress(self, nsamples=None):
         i = compress_weights(self.weights, self._rand, nsamples)
         return self.repeat(i)
 
+    @property
+    def _rand(self):
+        """Random number for consistent compression."""
+        seed = hash_pandas_object(self.index).sum() % 2**32
+        with temporary_seed(seed):
+            return np.random.rand(len(self))
+
+    @property
+    def _weighted(self):
+        """Determine if weights are actually present."""
+        return 'weights' in self.index.names
```

Comment:
* This doesn't need to be private, right? We could call it `isweighted`, which is the typical way of indicating a boolean return.
   (Same question for `WeightedDataFrame`)
* I think I would prefer this to be a method rather than a property, since it is a method for `WeightedDataFrame`.

Reply:
Agreed. Done